### PR TITLE
Optimized createListFromArrayLike

### PR DESCRIPTION
**Pull request by kisbg**
Signed-off-by: bence gabor kis <kisbg@inf.u-szeged.hu>

### CODE REVIEW

File: src/runtime/Object.cpp [hunk 1]
```diff
@@ -1299,11 +1299,9 @@ ValueVector Object::createListFromArrayLike(ExecutionState& state, Value obj, ui
     uint64_t index = 0;
     //8. Repeat while index < len
     while (index < len) {
-        // a. Let indexName be ToString(index).
-        auto indexName = Value(index).toString(state);
         // b. Let next be Get(obj, indexName).
         // c. ReturnIfAbrupt(next).
-        auto next = o->get(state, ObjectPropertyName(state, indexName)).value(state, o);
+        auto next = o->get(state, ObjectPropertyName(state, index)).value(state, o);
```
Comment (ksh8281) [inline src/runtime/Object.cpp:1304]: Good idea!
How about to use `getIndexedProperty` here?

**Comment by clover2123** [comment]
@kisbg LGTM
But would you please revert `tools/test/test262/excludelist.orig.xml` file?
Each TC with "SLOW" tag means PASSED but runs slowly.
We run slow TCs in release mode only to reduce the test running time in Jenkins.

**Comment by kisbg** [comment]
@clover2123  i removed the corresponding test-cases in `tools/test/test262/template.xml`.  Since this pull request greatly decreased the run time on this test-cases. 
For Example:
`RegExp/CharacterClassEscapes/character-class-non-digit-class-escape-flags-u.js`

On master:
`real	0m17,437s
user	0m17,407s`

With this pull request:
`real	0m1,468s
user	0m1,352s`

**Comment by ksh8281** [comment]
Great!

**Pull request merged**